### PR TITLE
Parse ebpf event in one step

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
We now make ebpf events 4 bytes smaller and parse them in one rush.

#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by JAORMX — COMMENTED** [review]
Shouldn't we instead be trying to derive the number to use from `/proc/sys/kernel/pid_max`? Seems to me that using u32 might be error prone 😕

**Comment by saschagrunert** [comment]
> Shouldn't we instead be trying to derive the number to use from `/proc/sys/kernel/pid_max`? Seems to me that using u32 might be error prone confused

The maximum amount of pids on 64 bit systems is `2^22`, which cannot be larger than `2^32-1` (`u32` max).

**Comment by JAORMX** [comment]
> > Shouldn't we instead be trying to derive the number to use from `/proc/sys/kernel/pid_max`? Seems to me that using u32 might be error prone confused
> 
> The maximum amount of pids on 64 bit systems is `2^22`, which cannot be larger than `2^32-1` (`u32` max).

That makes sense. Thanks for replying!

File: internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c [hunk 1]
```diff
@@ -38,7 +38,7 @@ struct {
 } events SEC(".maps");
 
 struct event_t {
-    u64 pid;
+    u32 pid;
```
Comment (jhrozek) [inline internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c:41]: IIRC `pid_t` is a 32bit type so this change is correct, but I wonder (just for my curiosity, really), where does `u32` come from? It took me a bit to realize I'm not looking at `uint32_t` and I was wondering why we were not using `pid_t` in the first place here.

Is the type a BPF type?

Comment (saschagrunert) [reply internal/pkg/daemon/bpfrecorder/bpf/recorder.bpf.c:41]: Yes, `vmlinux.h` provides typedefs for them: https://github.com/kubernetes-sigs/security-profiles-operator/blob/main/internal/pkg/daemon/bpfrecorder/vmlinux/x86/6.0.7-301.fc37.x86_64.vmlinux.h

**Review by jhrozek — APPROVED** [review]
/lgtm

**Pull request merged**